src/System.Linq.Expressions/src/System/Linq/Expressions/NewExpression.cs

```diff
@@ -183,6 +183,8 @@ public static NewExpression New(ConstructorInfo constructor, IEnumerable<Express
         public static NewExpression New(ConstructorInfo constructor, IEnumerable<Expression> arguments, IEnumerable<MemberInfo> members)
         {
             ContractUtils.RequiresNotNull(constructor, nameof(constructor));
+            ContractUtils.RequiresNotNull(constructor.DeclaringType, nameof(constructor) + "." + nameof(constructor.DeclaringType));
```

Comment:
It feels a bit strange to add code that is not testable, but lets keep it for consistency with .NET fx